api4/emoji.go

```diff
@@ -156,3 +158,41 @@ func getEmojiImage(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Cache-Control", "max-age=2592000, public")
 	w.Write(image)
 }
+
+func searchEmojis(c *Context, w http.ResponseWriter, r *http.Request) {
+	emojiSearch := model.EmojiSearchFromJson(r.Body)
+	if emojiSearch == nil {
+		c.SetInvalidParam("term")
+		return
+	}
+
+	if emojiSearch.Term == "" {
+		c.SetInvalidParam("term")
+		return
+	}
+
+	emojis, err := c.App.SearchEmoji(emojiSearch.Term, emojiSearch.PrefixOnly, PER_PAGE_MAXIMUM)
+	if err != nil {
+		c.Err = err
+		return
+	} else {
+		w.Write([]byte(model.EmojiListToJson(emojis)))
+	}
+}
+
+func autocompleteEmojis(c *Context, w http.ResponseWriter, r *http.Request) {
+	name := r.URL.Query().Get("name")
+
+	if name == "" {
+		c.SetInvalidUrlParam("name")
+		return
+	}
+
+	emojis, err := c.App.SearchEmoji(name, true, 100)
```

Comment:
How about adding constant to explicitly mention the max return value of 100?

Reply:
Yeah, probably a good idea